### PR TITLE
Annotate System.IO.FileSystem.AccessControl for nullable

### PR DESCRIPTION
**Pull request by buyaa-n**
Contributes to https://github.com/dotnet/runtime/issues/2339
Most of the files were already annotated. There were no update in ref assembly, so didn't enabled nullability for ref project

### CODE REVIEW

File: src/libraries/System.IO.FileSystem.AccessControl/src/System/Security/AccessControl/FileSystemSecurity.cs [hunk 1]
```diff
@@ -14,23 +14,23 @@ public abstract class FileSystemSecurity : NativeObjectSecurity
         private const ResourceType s_ResourceType = ResourceType.FileObject;
 
         internal FileSystemSecurity(bool isContainer)
-            : base(isContainer, s_ResourceType, _HandleErrorCode, isContainer)
+            : base(isContainer, s_ResourceType, _HandleErrorCode!, isContainer)
```
Comment (buyaa-n) [inline src/libraries/System.IO.FileSystem.AccessControl/src/System/Security/AccessControl/FileSystemSecurity.cs]: I think I need to update the delegate return value into nullable, then this bang will no needed PR for that https://github.com/dotnet/runtime/pull/31781:

**Comment by stephentoub** [comment]
> There were no update in ref assembly, so didn't enabled nullability for ref project

You need to.  If you don't, anyone consuming these APIs will see them as oblivious.

**Review by eiriktsarpalis — APPROVED** [review]
LGTM, thanks

File: src/libraries/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj [hunk 2]
```diff
@@ -5,6 +5,7 @@
     <GeneratePlatformNotSupportedAssemblyMessage Condition="'$(TargetsWindows)' != 'true'">SR.PlatformNotSupported_AccessControl</GeneratePlatformNotSupportedAssemblyMessage>
     <Configurations>net461-Windows_NT-Debug;net461-Windows_NT-Release;$(NetCoreAppCurrent)-Windows_NT-Debug;$(NetCoreAppCurrent)-Windows_NT-Release;$(NetFrameworkCurrent)-Windows_NT-Debug;$(NetFrameworkCurrent)-Windows_NT-Release;netstandard2.0-Debug;netstandard2.0-Release;netstandard2.0-Windows_NT-Debug;netstandard2.0-Windows_NT-Release</Configurations>
     <Nullable>annotations</Nullable>
```
Comment (stephentoub) [inline src/libraries/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj]:
```suggestion
```

Comment (stephentoub) [reply src/libraries/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj]: The fact that this had `<Nullable>annotations</Nullable>` suggests there were files this was pulling in that were `#nullable enable`'d as part of annotating another library.  Can those files have their `#nullable enable`s removed now that this library is enabled as well?

**Review by stephentoub — APPROVED** [review]
Other than the comments to be addressed, LGTM.

File: src/libraries/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj (hunk 2 above)
Comment (buyaa-n) [reply src/libraries/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj]: I thought it was for adding Nullable attributes reference for lower environments, no? CC @eiriktsarpalis

Comment (buyaa-n) [reply src/libraries/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj]: Never mind I misunderstood 😝, will remove `<Nullable>annotations</Nullable>`, but not sure about #nullable enable as they still might been used somewhere ...

Comment (buyaa-n) [reply src/libraries/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj]: They are used in Microsoft.IO.Redist, so will leave them for now

**Pull request merged**